tictactoe-game/script.js

```diff
@@ -1,50 +1,192 @@
+	var allElement=document.getElementsByClassName("sq");
+	for(var i =0;i<allElement.length;i++){
+		allElement[i].style.color=null;
+		allElement[i].innerHTML=null;
+	}
+	
+	moves = 0;
+	document.getElementById('message').innerHTML=null;
+	clearArray();
+}
+
+/**
+	* Define the function to clear diagonals values container 
+*/
+function clearArray(){
+	flag=0;
+	diagonals1.splice(0,diagonals1.length);
+	diagonals2.splice(0,diagonals2.length);
+	diagonal1ClassName.splice(0,diagonal1ClassName.length);
+	diagonal2ClassName.splice(0,diagonal2ClassName.length);
+}
 
 function play(square) {
-	if(square.innerHTML==='') {
+	
+	if( square.innerHTML==='' && flag!=1) {
 		moves++;
 
-		if(player===1) {
+		if( player===1 ) {
 			square.innerHTML = 'X';
 			player = 2;
-		}
-		else {
+		}else {
 			square.innerHTML = 'O';
 			player = 1;
 		}
 		
-		var result = checkGameOver();
+		var result = winner();
 		
 		if( result!==false ) {
-			document.getElementById('message').innerHTML = 'Game Over: '+result;
-		}
-		else {
+			document.getElementById('message').innerHTML = 'Game Over: ' + result;
+			flag=1;
+		}else {
 			document.getElementById('message').innerHTML = 'Keep playing';
 		}
 	}
 }
 
-function checkGameOver() {
-	// 1st: check if theres less 5 moves
-	if(moves<5) {
-		return false;
+/**
+	*Define the function that check whether the board empty
+	*@return {isEmpty<boolean>}
+*/
+
+function isEmpty(rowColArray){
+	var flag = true ;
+	for(var i=0;i<rowColArray.length;i++){
+		if(rowColArray[i].innerHTML==='')
+			return true;	
+		else 
+			flag=false;	
 	}
-	
-	// 2nd: check for winner lines
-	var row1 = document.getElementsByClassName('r1');
-	
-	if( row1[0].innerHTML!=='' &&
-			row1[0].innerHTML===row1[1].innerHTML &&
-			row1[0].innerHTML===row1[2].innerHTML) {
-		return row1[0].innerHTML + ' is the winner.';
+	return flag;
+}
+
+/**
+	*Define the function which set winner to Red
+	*@param {markWinner<Array, boolean>}
+*/
+function markWinner(elementArray,flag){
+	/**
+		* Access rows , columns and diagonals element and set color to Red
+		*@type {diagonalClass <element>}
+	*/
+	var diagonalClass;
+	for(var i=0; i<elementArray.length; i++){
+		if(!flag){
+			elementArray[i].style.color = "red";
+		}else {
+
+			/** get element by class name from the array passed by reference*/
+			diagonalClass = document.getElementsByClassName(elementArray[i]);
+			
+			/** Style each element identified by class name  row and column*/
+			diagonalClass[0].style.color='red';
+		}
+
 	}
-	
-	// 3rd: check if board is full
-	if(moves>8){
-		return "It's a tie";
+}
+
+function winner(){
+
+/**
+	* Players move untill Five turn 
+	* @return {<boolean>}
+*/	
+	if( moves<5 ) {
+		return false;
 	}
-	// continue game
+	
+/**
+	*	Clear the diagonals and arrays containing class names, for fresh value 
+*/
+	clearArray();
+
+/**
+	*Check each rows , columns and Diagonals are match
+	*Style the matching line to RED
+	*@return {<string>} 
+*/
+	for(var row=0;row<3;row++){
```

Comment:
I'm not sure if you guys are allowed to use `Array.map()` or `Array.filter()` for this assignment, but they would both make this markup much more easy to follow. Even if you can't use them, you should rewrite this with them and see what you come up with.